R/Google.R

```diff
@@ -0,0 +1,207 @@
+          cases_total = as.numeric(.data$cases_total),
+          deaths_new = as.numeric(.data$deaths_new),
+          deaths_total = as.numeric(.data$deaths_total),
+          recovered_new = as.numeric(.data$recovered_new),
+          recovered_total = as.numeric(.data$recovered_total),
+          tested_new = as.numeric(.data$tested_new),
+          tested_total = as.numeric(.data$tested_total),
+          hosp_new = as.numeric(.data$hosp_new),
+          hosp_total = as.numeric(.data$hosp_total)
+        ) %>%
+        replace_na(
+          list(
+            level_1_region = "Unknown",
+            level_2_region = "Unknown"
+          )
+        ) %>%
+        select(
+          date,
+          level_1_region,
+          level_1_region_code,
+          level_2_region,
+          level_2_region_code,
+          cases_new,
+          cases_total,
+          deaths_new,
+          deaths_total,
+          recovered_new,
+          recovered_total,
+          hosp_new,
+          hosp_total,
+          tested_new,
+          tested_total,
+          everything()
+        )
+    },
+
+    #' @description Google specific subregion level data cleaning. Takes the
+    #' data cleaned by `clean_common` and aggregates it to the country level
+    #' (level 1).
+    #' @importFrom dplyr select summarise group_by across everything
+    #' @importFrom rlang .data
+    clean_level_1 = function() {
+      self$data$clean <- self$data$clean %>%
+        group_by(
+          .data$date, .data$level_1_region_code,
+          .data$level_1_region
+        ) %>%
+        summarise(across(where(is.double), sum))
+    },
+
+    #' @description Google specific subregion2 level data cleaning. Takes the
+    #' data cleaned by `clean_common` and aggregates it to the subregion level
+    #' (level 2).
+    #' @importFrom dplyr select summarise group_by across everything
+    #' @importFrom rlang .data
+    clean_level_2 = function() {
+      self$data$clean <- self$data$clean %>%
+        group_by(
+          .data$date, .data$level_1_region_code, .data$level_1_region,
+          .data$level_2_region_code, .data$level_2_region
+        ) %>%
+        summarise(across(where(is.double), sum))
+    },
+    #' @description custom initialize for Google
+    #' @param ... arguments to be passed to `DataClass` and initialize Google
+    initialize = function(...) {
+      super$initialize(...)
+      if (self$level == "3" & is.null(self$target_regions)) {
+        msg <- paste(
+          "Processing google covid-19 data at level 3 with no target",
+          "regions will take a long time and contain data for multiple",
+          "countries. Consider not running the 'process' step or filtering",
+          "to a specific country using the 'regions' argument."
+        )
+        message_verbose(self$verbose, msg)
+      }
+    },
+
+    #' @description display the countries available with this data
+    show_countries = function() {
```

Comment:
we are copy pasting this code into both JHU and Google. I think this means it might be time for another parent class that inherits fro CountryDataClass. Maybe ComplexCountryDataClass or CountryRegionalDataClass (😆 )